leo_fw/src/leo_fw/test_hw.py

```diff
@@ -0,0 +1,385 @@
+from enum import Enum
+
+import time
+
+import rospy
+import rosnode
+import rospkg
+
+import yaml
+
+from leo_msgs.msg import Imu, WheelStates
+from geometry_msgs.msg import Twist
+from std_msgs.msg import Float32
+from .utils import write_flush
+from .board import BoardType, determine_board, check_firmware_version
+
+
+def parse_yaml(file_path):
+    with open(file_path, "r") as stream:
+        try:
+            return yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+    return {}
+
+
+class TestMode(Enum):
+    ENCODER = "encoder"
+    TORQUE = "torque"
+    IMU = "imu"
+    BATTERY = "battery"
+    ALL = "all"
+
+    def __str__(self):
+        return self.value
+
+
+class HardwareTester:
+
+    is_new_imu_data = 0
+    is_new_wheel_data = 0
+    is_new_battery_data = 0
+    is_wheel_loaded = 1
+
+    imu_data = Imu()
+    wheel_data = WheelStates()
+    battery_data = Float32()
+
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    ENDC = "\033[0m"
+
+    def __init__(self):
+
+        self.rospack = rospkg.RosPack()
+        self.path = self.rospack.get_path("leo_fw") + "/test_data/"
+
+        ###Publisher
+
+        self.cmd_vel_pub = rospy.Publisher("cmd_vel", Twist, queue_size=1)
+        self.cmd_pwmfl_pub = rospy.Publisher(
+            "firmware/wheel_FL/cmd_pwm_duty", Float32, queue_size=1
+        )
+        self.cmd_pwmrl_pub = rospy.Publisher(
+            "firmware/wheel_RL/cmd_pwm_duty", Float32, queue_size=1
+        )
+        self.cmd_pwmfr_pub = rospy.Publisher(
+            "firmware/wheel_FR/cmd_pwm_duty", Float32, queue_size=1
+        )
+        self.cmd_pwmrr_pub = rospy.Publisher(
+            "firmware/wheel_RR/cmd_pwm_duty", Float32, queue_size=1
+        )
+        self.cmd_velfl_pub = rospy.Publisher(
+            "firmware/wheel_FL/cmd_velocity", Float32, queue_size=1
+        )
+        self.cmd_velrl_pub = rospy.Publisher(
+            "firmware/wheel_RL/cmd_velocity", Float32, queue_size=1
+        )
+        self.cmd_velfr_pub = rospy.Publisher(
+            "firmware/wheel_FR/cmd_velocity", Float32, queue_size=1
+        )
+        self.cmd_velrr_pub = rospy.Publisher(
+            "firmware/wheel_RR/cmd_velocity", Float32, queue_size=1
+        )
+
+        ###Subscriber
+
+        self.battery_sub = rospy.Subscriber(
+            "firmware/battery", Float32, self.battery_callback
+        )
+        self.wheel_sub = rospy.Subscriber(
+            "firmware/wheel_states", WheelStates, self.wheel_callback
+        )
+        self.imu_sub = rospy.Subscriber("firmware/imu", Imu, self.imu_callback)
+
+    def __del__(self):
+        pass
+
+    def battery_callback(self, data):
```

Comment:
Add type annotations for method arguments and return types. For example, this line should be changed to:
```Python
def battery_callback(self, data: Float32) -> None:
```